### PR TITLE
Use new build image 0.20.3

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

The new build image uses the latest version **v1.45.2** of golangci-lint. With this version running the `make lint` locally yields the same results as running on CI.

**Special notes for your reviewer**:

* Requires https://github.com/grafana/loki/pull/5924 so that `0.20.3` is built
* ~~This PR contains commits from https://github.com/grafana/loki/pull/5901~~
* The changes in file `.drone/drone.yml` were generated by `make drone`
* Code changes are fixes for the linter:
  ```
   pkg/storage/stores/tsdb/index/index.go:1525:61: unexported-return: exported func ReadFingerprintOffsetsTable returns unexported type index.fingerprintOffsets, which can be annoying to use (revive)
   func ReadFingerprintOffsetsTable(bs ByteSlice, off uint64) (fingerprintOffsets, error) {
                                                            ^
   pkg/ingester/ingester.go:554:59: unexported-return: exported method GetOrCreateInstance returns unexported type *ingester.instance, which can be annoying to use (revive)
   func (i *Ingester) GetOrCreateInstance(instanceID string) *instance {
   ```

### CODE REVIEW

File: .drone/drone.yml [hunk 1]
```diff
@@ -1,143 +1,171 @@
 ---
 kind: pipeline
 name: loki-build-image
+
+platform:
```
Comment (jeschkies) [inline .drone/drone.yml:5]: Interesting. I would have expected just the versions to change.

**Review by jeschkies — COMMENTED** [review]
Ok, according to the [build **logs](https://drone.grafana.net/grafana/loki/10832/1/2https://drone.grafana.net/grafana/loki/10832/1/2)** of you build image change. It pushed `grafana/loki-build-image:0.20.2`

And I know why. You didn't regenerate the `drone.yml` :see_no_evil: `make check-drone-drift` was supposed to catch that.

File: Makefile [hunk 2]
```diff
@@ -27,7 +27,7 @@ DOCKER_IMAGE_DIRS := $(patsubst %/Dockerfile,%,$(DOCKERFILES))
 BUILD_IN_CONTAINER ?= true
 
 # ensure you run `make drone` after changing this
-BUILD_IMAGE_VERSION := 0.20.1
+BUILD_IMAGE_VERSION := 0.20.3
```
Comment (jeschkies) [inline Makefile:30]: Since you recent change just overrode `0.20.2` it should be that
```suggestion
BUILD_IMAGE_VERSION := 0.20.2
```

Comment (chaudum) [reply Makefile:30]: 0.20.2 was built but never been used, see https://github.com/grafana/loki/commit/435391661f4e0f03eb378f77c07aaad5009cd6e9#diff-60045f9474705196e786df20a7bb40d29e013e47a3b6927153a10b9acac82266L358

File: pkg/ingester/ingester.go [hunk 3]
```diff
@@ -551,7 +551,7 @@ func (i *Ingester) Push(ctx context.Context, req *logproto.PushRequest) (*logpro
 	return &logproto.PushResponse{}, err
 }
 
-func (i *Ingester) GetOrCreateInstance(instanceID string) *instance {
+func (i *Ingester) GetOrCreateInstance(instanceID string) *instance { //nolint:revive
```
Comment (jeschkies) [inline pkg/ingester/ingester.go:554]: This change seems unrelated.

**Review by jeschkies — CHANGES_REQUESTED** [review]

File: pkg/ingester/ingester.go (hunk 3 above)
Comment (chaudum) [reply pkg/ingester/ingester.go:554]: The new linter version reports: `unexported-return: exported method GetOrCreateInstance returns unexported type *ingester.instance`

**Review by jeschkies — APPROVED** [review]
